hazelcast/src/hazelcast/client/spi.cpp

```diff
@@ -3071,17 +3072,21 @@ std::unordered_map<address, address>
 cloud_discovery::get_addresses()
 {
 #ifdef HZ_BUILD_WITH_SSL
+    std::string discovery_token{ cloud_config_.discovery_token };
     try {
+
         util::SyncHttpsClient httpsConnection(cloud_base_url_,
                                               std::string(CLOUD_URL_PATH) +
-                                                cloud_config_.discovery_token,
+                                                discovery_token,
                                               timeout_);
         auto& conn_stream = httpsConnection.connect_and_get_response();
         return parse_json_response(conn_stream);
     } catch (std::exception& e) {
+        std::string message{ e.what() };
+        boost::replace_all(message, discovery_token, "<DISCOVERY_TOKEN>");
         std::throw_with_nested(
```

Comment:
can it leak due to nested exception?

Reply:
On application crash it is not printed to `stdout`. But the token is included `nested_exception` that's why I added `filter` handler to `SyncHttpsRequest` earlier.